lib/ClangImporter/ClangImporter.cpp

```diff
@@ -742,33 +741,32 @@ ClangImporter::getPCHFilename(const ClangImporterOptions &ImporterOptions,
   llvm::sys::path::replace_extension(PCHBasename, ".pch");
   SmallString<256> PCHFilename { PCHOutputDir };
   llvm::sys::path::append(PCHFilename, PCHBasename);
-  return PCHFilename.str().str();
+  return Impl.SwiftContext.AllocateCopy(PCHFilename.str());
```

Comment:
This isn't being cached anywhere. Can you only context-allocate it when storing it in the SinglePCH box?